src/Enterspeed.Source.UmbracoCms/NotificationHandlers/EnterspeedContentCacheRefresherNotificationHandler.cs

```diff
@@ -176,6 +189,15 @@ public void Handle(ContentCacheRefresherNotification notification)
                 }
             }
 
+            if (_enterspeedMasterContentService.IsMasterContentEnabled())
```

Comment:
I think we need to walk through this again, for me to fully understand what is going on here. :)